<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EngineDesignerWidget</class>
 <widget class="QWidget" name="EngineDesignerWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>976</width>
    <height>544</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Engine Designer</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <widget class="QFrame" name="frame">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>460</width>
       <height>0</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>300</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <property name="lineWidth">
      <number>1</number>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QGroupBox" name="baseSpecsGroupBox">
        <property name="title">
         <string/>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QLabel" name="mEngineTypeLabel">
             <property name="text">
              <string>Engine Type</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="mEngineTypeLineEdit">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QLabel" name="mModelNameLabel">
             <property name="text">
              <string>Model Name</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="mModelNameLineEdit">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QLabel" name="mLengthLabel">
             <property name="text">
              <string>Length            </string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="mLengthLineEdit"/>
           </item>
           <item>
            <widget class="QLabel" name="mLengthFtLabel">
             <property name="text">
              <string>ft  </string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_5">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <item>
            <widget class="QLabel" name="mDiameterLabel">
             <property name="text">
              <string>Diameter        </string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="mDiameterLineEdit"/>
           </item>
           <item>
            <widget class="QLabel" name="mDiameterFtLabel">
             <property name="text">
              <string>ft  </string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_6">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_8">
           <item>
            <widget class="QLabel" name="mThrustLabel">
             <property name="text">
              <string>Thrust Offset </string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="mThrustOffsetLineEdit">
             <property name="toolTip">
              <string>Location of the thrust relative to the engineâ€™s reference point along the x-axis. A positive value is forward of the engine, negative is aft</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="mThrustOffsetFtLabel">
             <property name="text">
              <string>ft  </string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_7">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>Modify Plots</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_7">
         <item row="0" column="0">
          <layout class="QGridLayout" name="gridLayout_8">
           <item row="1" column="3">
            <spacer name="horizontalSpacer_9">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="mGlobalMultiplierLabel">
             <property name="text">
              <string>Global Multiplier</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QCheckBox" name="mShowModifierCheckbox">
             <property name="text">
              <string>Show Modifier Line</string>
             </property>
             <property name="checked">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="mGlobalMultiplierLineEdit"/>
           </item>
           <item row="1" column="2">
            <widget class="QPushButton" name="mApplyMultiplierButton">
             <property name="text">
              <string>Apply</string>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QPushButton" name="mApplyOffsetButton">
             <property name="text">
              <string>Apply</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="mGlobalOffsetButton">
             <property name="text">
              <string>Global Offset</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <layout class="QGridLayout" name="gridLayout_9">
             <item row="0" column="0">
              <widget class="QRadioButton" name="mOffsetRadioButton">
               <property name="text">
                <string>Offset</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QRadioButton" name="mMultiplierRadioButton">
               <property name="text">
                <string>Multiplier</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="2" column="1">
            <widget class="QLineEdit" name="mGlobalOffsetLineEdit"/>
           </item>
          </layout>
         </item>
         <item row="1" column="0">
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="0" column="0">
            <spacer name="horizontalSpacer_10">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="0" column="1">
            <widget class="QPushButton" name="mRevertPlotButton">
             <property name="text">
              <string> Revert Plot</string>
             </property>
             <property name="icon">
              <iconset resource="resources/icons.qrc">
               <normaloff>:/icons/Light/64x64/revert.png</normaloff>:/icons/Light/64x64/revert.png</iconset>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <spacer name="horizontalSpacer_4">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="mSaveModelButton">
          <property name="text">
           <string>Save Model</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="frame_2">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Current Plot:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="plotTypeComboBox"/>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Designer::PlotManagerWidget" name="plotManagerWidget"/>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Designer::PlotManagerWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>PlotManagerWidget.hpp</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>mEngineTypeLineEdit</tabstop>
  <tabstop>mModelNameLineEdit</tabstop>
  <tabstop>mLengthLineEdit</tabstop>
  <tabstop>mDiameterLineEdit</tabstop>
  <tabstop>mThrustOffsetLineEdit</tabstop>
  <tabstop>mGlobalMultiplierLineEdit</tabstop>
  <tabstop>mApplyMultiplierButton</tabstop>
  <tabstop>mGlobalOffsetLineEdit</tabstop>
  <tabstop>mApplyOffsetButton</tabstop>
  <tabstop>mShowModifierCheckbox</tabstop>
  <tabstop>mOffsetRadioButton</tabstop>
  <tabstop>mMultiplierRadioButton</tabstop>
  <tabstop>mRevertPlotButton</tabstop>
  <tabstop>mSaveModelButton</tabstop>
  <tabstop>plotTypeComboBox</tabstop>
 </tabstops>
 <resources>
  <include location="resources/icons.qrc"/>
 </resources>
 <connections/>
</ui>
